M2/Macaulay2/m2/examples.m2

```diff
@@ -170,10 +171,24 @@ extractExamples = docBody -> (
 -----------------------------------------------------------------------------
 
 examples = method(Dispatch => Thing)
-examples Hypertext := dom -> raise(stack extractExamplesLoop dom, -1)
+examples Hypertext := h -> (
+    ex := extractExamplesLoop h;
+    if #ex == 0 then "-- no examples" else stack ex)
 examples Thing     := key -> (
-    rawdoc := fetchAnyRawDocumentation makeDocumentTag key;
-    if rawdoc =!= null and rawdoc.?Description then examples DIV{rawdoc.Description})
+    tag := makeDocumentTag key;
+    rawdoc := fetchAnyRawDocumentation tag;
+    if rawdoc =!= null and rawdoc.?Description
+    then stack(
+	"-- examples for tag: " | format tag,
+	"-- " | net locate tag,
+	examples DIV{rawdoc.Description}))
+
+examples List := L -> (
+    L = sortBy(x -> x#1) \\ nonnull \\ examples \ L; -- sort by location
```

Comment:
Why are you reordering the list? The input could be something out of `about` or `methods`, and the order should also match that.

Reply:
To be clear, I understand that you're deduplicating the list, which is good, but not if the order is changed. Maybe you could do something similar to `dedupMethods` from `code.m2`?

Reply:
Yeah, `dedupMethods` was my inspiration for this. I went back and forth as to whether we should sort by location or not. If we leave them in the original order, then we'll still have some duplication. But it probably would be surprising to the user for things to get shuffled around! I'll remove the sorting.

Reply:
I think as long as the order of first appearance is the same, it shouldn't be a big deal. e.g. `methods res` gives `resolution(Ideal)`, then matrix and module, but then `resolution(MonomialIdeal)` has the same documentation, and if you wanted to merge `MonomialIdeal` with `Ideal` it would be fine, as long as the order `Ideal/MonomialIdeal Matrix Module` is the same. I do think sorting outputs of `about` by location of documentation would be reasonable, since it needs to pull all the documentation anyway.

Reply:
I went and and stopped sorting.  We only de-duplicate examples if they happened to be next to each other already in the original list.